packages/accordion/src/vaadin-accordion-panel.js

```diff
@@ -149,47 +148,26 @@ class AccordionPanel extends DetailsMixin(
   }
 
   /** @private */
-  _initSummary() {
-    this._summaryController = new SummaryController(this);
-    this.addController(this._summaryController);
-
-    // Wait for heading element render to complete
-    afterNextRender(this, () => {
-      this._toggleElement = this.focusElement.$.button;
-    });
+  _summaryChanged(summary) {
+    this._summaryController.setSummary(summary);
   }
 
   /** @private */
-  _initContent() {
-    this._contentController = new ContentController(this);
-    this._contentController.addEventListener('slot-content-changed', (event) => {
-      // Store nodes to toggle `aria-hidden` attribute
-      const content = event.target.nodes || [];
-      this._contentElements = content;
-
-      // See https://www.w3.org/WAI/ARIA/apg/patterns/accordion/
-      const node = content[0];
+  __updateAriaAttributes(focusElement, contentElements) {
+    if (focusElement && contentElements) {
+      const node = contentElements[0];
+
       if (node) {
         node.setAttribute('role', 'region');
-        node.setAttribute('aria-labelledby', this.focusElement.id);
+        node.setAttribute('aria-labelledby', focusElement.id);
       }
 
-      if (node && node.parentNode === this && node.id) {
```

Comment:
The check for `parentNode` is now moved to setting `_contentElements` in `DetailsMixin`.